huaweicloud/services/rds/resource_huaweicloud_rds_mysql_binlog.go

```diff
@@ -0,0 +1,204 @@
+	instanceId := d.Id()
+	getMysqlBinlogPath := getMysqlBinlogClient.Endpoint + getMysqlBinlogHttpUrl
+	getMysqlBinlogPath = strings.ReplaceAll(getMysqlBinlogPath, "{project_id}", getMysqlBinlogClient.ProjectID)
+	getMysqlBinlogPath = strings.ReplaceAll(getMysqlBinlogPath, "{instance_id}", instanceId)
+	getMysqlBinlogOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+		MoreHeaders: map[string]string{"Content-Type": "application/json"},
+	}
+	getMysqlBinlogResp, err := getMysqlBinlogClient.Request("GET", getMysqlBinlogPath, &getMysqlBinlogOpt)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	getMysqlBinlogRespBody, err := utils.FlattenResponse(getMysqlBinlogResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	binlog := utils.PathSearch("binlog_retention_hours", getMysqlBinlogRespBody, nil)
+
+	if binlog == nil {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "")
+	}
+	mErr = multierror.Append(
+		mErr,
+		d.Set("region", region),
+		d.Set("instance_id", instanceId),
+		d.Set("binlog_retention_hours", binlog),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func resourceMysqlBinlogDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	var (
+		deleteMysqlBinlogHttpUrl = "v3/{project_id}/instances/{instance_id}/binlog/clear-policy"
+		deleteMysqlBinlogProduct = "rds"
+	)
+	deleteMysqlBinlogClient, err := cfg.NewServiceClient(deleteMysqlBinlogProduct, region)
+	if err != nil {
+		return diag.Errorf("error deleting RDS client: %s", err)
+	}
+	instanceId := d.Get("instance_id").(string)
+	deleteMysqlBinlogPath := deleteMysqlBinlogClient.Endpoint + deleteMysqlBinlogHttpUrl
+	deleteMysqlBinlogPath = strings.ReplaceAll(deleteMysqlBinlogPath, "{project_id}",
+		deleteMysqlBinlogClient.ProjectID)
+	deleteMysqlBinlogPath = strings.ReplaceAll(deleteMysqlBinlogPath, "{instance_id}", instanceId)
+	deleteMysqlBinlogOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+	log.Printf("[DEBUG] Delete RDS Mysql binlog options: %#v", deleteMysqlBinlogOpt)
+	requestBody := map[string]interface{}{
+		"binlog_retention_hours": 0,
+	}
+	deleteMysqlBinlogOpt.JSONBody = requestBody
+	retryFunc := func() (interface{}, bool, error) {
+		_, err = deleteMysqlBinlogClient.Request("PUT", deleteMysqlBinlogPath, &deleteMysqlBinlogOpt)
+		retry, err := handleMultiOperationsError(err)
+		return nil, retry, err
+	}
+	_, err = common.RetryContextWithWaitForState(&common.RetryContextWithWaitForStateParam{
+		Ctx:          ctx,
+		RetryFunc:    retryFunc,
+		WaitFunc:     rdsInstanceStateRefreshFunc(deleteMysqlBinlogClient, instanceId),
+		WaitTarget:   []string{"ACTIVE"},
+		Timeout:      d.Timeout(schema.TimeoutDelete),
+		DelayTimeout: 1 * time.Second,
+		PollInterval: 10 * time.Second,
+	})
+	if err != nil {
+		return diag.Errorf("error deleting RDS Mysql binlog: %s", err)
+	}
+	d.SetId(instanceId)
+	return nil
```

Comment:
```suggestion
```
unnecessary

Reply:
A return value is required and cannot be removed.